### ISSUE TITLE
Unchecked _max_tb causing TypeError: '<=' not supported between instances of 'int' and 'nonetype'

### ISSUE THREAD

**Issue by GPhilo** [okken/pytest-check#137]
There seems to be a missing check in [this line](https://github.com/okken/pytest-check/blob/c7e7741e4d5665a07b0985932acc484aac2d5095/src/pytest_check/check_log.py#L51C11-L51C37) causing a TypeError inside `log_failure`.

Shouldn't this be checking for `_max_tb` as well before the <=, like:
```python
if (_max_tb is None) or (_num_failures <= _max_tb):
# rest of code
```

**Comment by okken** [comment]
Nice catch. It seems like it, looking at the code, since _max_tb defaults to None in this file.
However, in use, it is always set. 
I can't come up with a use case to make this fail line raise an exception as is. 

I think the fix may be to set the default value to 1 instead of None, since it more matches the current implementation.
Thanks for reviewing the code.

**Comment by GPhilo** [comment]
Haha, I don't know if I'm using the code in an unexpected way then, but for me that was still initialized to None and causing issues 😄

**Comment by okken** [comment]
Do you have a toy example that causes the failure?

**Comment by okken** [comment]
I'll fix it, but I'd like to have a way to reproduce the problem

**Comment by GPhilo** [comment]
I'm going on holiday tomorrow for a week, but once I'm back home I'll try to come up with some code to reproduce the issue (just so you know why my next comment will take a while).

**Comment by okken** [comment]
I've got a fix, I'll push in place soon

**Comment by okken** [comment]
Fixed with 2.2.2
